File: listas/Listasimple.java
import java.util.*;

class Listasimple {
    public int i;
    Scanner entrada;
    public Listasimple sig;

    Listasimple() {
        i = 0;
        sig = null;
        entrada = new Scanner(System.in);
    }

    public void crearLista() {
        Listasimple lista = null;
        Listasimple aux = null;
        System.out.println("Dame el tamaño de la lista");
        int n = entrada.nextInt();
        for (int a = 0; a < n; a++) {
            if (lista == null) {
                lista = new Listasimple();
                System.out.println("Dame el valor de i");
                lista.i = entrada.nextInt();
                aux = lista;
            } else {
                aux.sig = new Listasimple();
                aux = aux.sig;
                System.out.println("Dame el valor de i");
                aux.i = entrada.nextInt();
            }
        }
        imprimirLista(lista);
    }

    public void imprimirLista(Listasimple l) {
        Listasimple aux = l;
        while (aux != null) {
            System.out.println(aux.i);
            aux = aux.sig;
        }
    }

    public static void main(String[] args) {
        Listasimple llamar = new Listasimple();
        llamar.crearLista();
    }
}

File: listas/ListaSimple.cpp
#include <iostream>
using namespace std;
class ListaSimple{
    public:
    int i;
    ListaSimple *sig;
    void crearLista();
    void imprimirLista(ListaSimple *l);
};
void ListaSimple::crearLista(){
    int n;
    ListaSimple *lista=NULL,*aux;
    cout<<"De cuantos nodos deseas la lista"<<endl;
    cin>>n;

    for(int a=0;a<n;a++){
        if(lista==NULL){
            lista = new ListaSimple();
            cout<<"Dame el valor de i"<<endl;
            cin>>lista->i;
            aux = lista;
        }else{
            aux->sig=new ListaSimple();
            aux = aux->sig;
            cout<<"Dame el valor de i"<<endl;
            cin>>aux->i;
        }
    }//llave del for
    imprimirLista(lista);
}
void ListaSimple::imprimirLista(ListaSimple *l){
    ListaSimple *aux = l;
    while(aux!=NULL){
        cout<<aux->i<<endl;
        aux = aux->sig;clear
    }
}
int main(){
    ListaSimple *llamar = new ListaSimple();
    llamar->crearLista();
}